### PR TITLE
Pass query string params to interactive dialog request url

### LINKED ISSUE #14347: Query string parameters are omitted from interactive dialog request url
When the server is routing an interactive dialog url using a relative path ('i.e /plugins/{pluginId}/api/myEndpoint), it will unintentionally drop the query strings parameters when executing the request in https://github.com/mattermost/mattermost-server/blob/84859e71597a5f0f02a77e99f7a280efd0ed840d/app/integration_action.go#L328

The query string parameters should be relayed back to the plugin with no modifications.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-24024

### LINKED ISSUE #24024: [Bug] Cannot create slash command
#### Summary
Can't create slash command after update version 7=>8 - white window and no more.

#### Steps to reproduce
Open Integrations - Slash commands - Create - See white window

#### Expected behavior
Show create slash command window

#### Observed behavior (that appears unintentional) 
image
image
```JavaScript Console Error
Error: [@formatjs/intl] An `id` must be provided to format a message. You can either:
1. Configure your build toolchain with [babel-plugin-formatjs](https://formatjs.io/docs/tooling/babel-plugin)
or [@formatjs/ts-transformer](https://formatjs.io/docs/tooling/ts-transformer) OR
2. Configure your `eslint` config to include [eslint-plugin-formatjs](https://formatjs.io/docs/tooling/linter#enforce-id)
to autofix this issue
    at p (utils.js:76:15)
    at ct (message.js:31:5)
    at wt (provider.js:46:18)
    at Wt (message.js:21:17)
    at ao (react-dom.production.min.js:157:137)
    at $u (react-dom.production.min.js:267:460)
    at Ni (react-dom.production.min.js:250:347)
    at _i (react-dom.production.min.js:250:278)
    at Ci (react-dom.production.min.js:250:138)
    at vi (react-dom.production.min.js:243:163)
```

#### Possible fixes
Reload page, restart application

Current version is 8.0.0 (master/Tue Jul 11 13:17:40 UTC 2023) (Docker mattermost/mattermost-enterprise-edition:master)

### PR DESCRIPTION
**Pull request by Ashniu123**
Closes #14347 on Github
Closes #24024 on Jira

#### Summary
Query string parameters are now passed onto the interactive dialog request url

#### Ticket Link
Github issue: https://github.com/mattermost/mattermost-server/issues/14347
Jira: https://mattermost.atlassian.net/browse/MM-24024

### CODE REVIEW

**Comment by Ashniu123** [comment]
I do not know why the "MySQL test cases" are failing. I think my changes have nothing to do with that.
@marianunez What do you think?

**Comment by marianunez** [comment]
@Ashniu123 Yes, the failing build does not seem related to your PR. I'll take a look to see why it's not passing properly.

**Comment by Ashniu123** [comment]
I could but I will need some guidance.
How do I mock the function and know what parameters are passed into it when the function is called?
Could you give a reference example file in the codebase?

**Comment by lieut-data** [comment]
Hey @Ashniu123! I'd suggest modifying `TestSubmitInteractiveDialog` in `app/integration_action_test.go` to verify both the third-party integrations (via HTTP) and the plugins get the query string parameters. To verify the plugin case, you'll see the inline code block there that will be compiled: you'll want to check the incoming `http.Request`, and if the expected values aren't there, write something else to the `http.ResponseWriter`, so that the "outside" assertion fails appropriately.

**Comment by lieut-data** [comment]
Thanks, @Ashniu123 -- added to my list to review :)

**Comment by Ashniu123** [comment]
This PR has been idle on open for too long with no action. Can we get this done asap please?

**Comment by lieut-data** [comment]
@Ashniu123, heads up that the tests are currently failing with:
```
    TestSubmitInteractiveDialog: integration_action_test.go:546: 
        	Error Trace:	integration_action_test.go:546
        	Error:      	Not equal: 
        	            	expected: "some other error"
        	            	actual  : "some error"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-some other error
        	            	+some error
        	Test:       	TestSubmitInteractiveDialog
```

**Comment by Ashniu123** [comment]
Must have happened due to the [latest commit](https://github.com/mattermost/mattermost-server/pull/14366/commits/5b1315e417773d0ddfe244145f23af3873ea9beb). It was working fine until then.

Please be careful about the changes @lieut-data 

You added an extra '2' to "xyz" as depicted [here](https://github.com/mattermost/mattermost-server/pull/14366/commits/5b1315e417773d0ddfe244145f23af3873ea9beb#diff-becd7985a7ca868cf0457fe230487584R494)

File: app/integration_action_test.go [hunk 1]
```diff
@@ -489,12 +489,12 @@ func TestSubmitInteractiveDialog(t *testing.T) {
...
 
-		func (p *MyPlugin) 	ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
-				errReply := "some error"
-				if r.URL.Query().Get("abc") == "xyz" {
-					errReply = "some other error"
-				}
-		    response := &model.SubmitDialogResponse{
+		func (p *MyPlugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+			errReply := "some error"
+			if r.URL.Query().Get("abc") == "xyz2" {
```
Comment (lieut-data) [inline app/integration_action_test.go:494]:
```suggestion
 			if r.URL.Query().Get("abc") == "xyz" {
```

**Comment by ashishbhate** [comment]
I think that ping was meant for @Ashniu123 :)

**Comment by lieut-data** [comment]
Hah, well this PR is turning out to be a comedy of errors on my part! Thanks for your patience, @Ashniu123 and @ashishbhate :D

**Comment by teresa-novoa** [comment]
@DHaussermann friendly reminder to review this PR, thanks!

**Comment by DHaussermann** [comment]
Hi All, Apologies for not getting to this PR much sooner.  

I'm not entirely sure how to validate this. @lieut-data or @marianunez can you provide tests steps for this? Would I use a curl or Postman request to test this or is there a plugin that would expose this issue?

**Comment by lieut-data** [comment]
@DHaussermann, ideally we'd make a change to the demo plugin to effect this. I think the unit tests largely satisfy the testing requirements here, though!

**Comment by DHaussermann** [comment]
Thanks for clarifying @lieut-data. 
I'll follow-up to add an enhancement request to the demo plugin repo that would make this more piratical to test.

**Pull request merged**